sql/plan/subquery.go

```diff
@@ -221,7 +221,7 @@ func (s *Subquery) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 func prependRowInPlan(row sql.Row) func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
 	return func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
 		switch n := n.(type) {
-		case sql.Table, *Project, *GroupBy, *Window, *Having, *ValueDerivedTable:
+		case sql.Table, *Project, *GroupBy, *Window, *ValueDerivedTable:
```

Comment:
Great catch! You're totally right that `*Having` shouldn't be in this list. 

I think you can further simplify this a little more with the new `sql.Projector` interface I just added: 

```suggestion
		case sql.Table, sql.Projector, *ValueDerivedTable:
```